multiplex_immuno_processing/generate_alignment_paramters.py

```diff
@@ -1,389 +1,98 @@
-        )
-        offset_list.append(meanoffset)
-
-    return offset_list
-
-
-def find_xyz_offset(target_img, test_img, ploton=False, verbose=False):
-    import skimage.exposure as skex
-
-    test_img_rs = test_img.copy()
-    target_img_8 = skex.rescale_intensity(
-        target_img.copy(), in_range="image", out_range="uint8"
-    ).astype("uint8")
-    test_img_8 = skex.rescale_intensity(
-        test_img_rs.copy(), in_range="image", out_range="uint8"
-    ).astype("uint8")
-    target_slice = target_img_8.copy().max(axis=0)
-    test_slice = test_img_8.copy().max(axis=0)
-
-    if ploton:
-        fig, axr = plt.subplots(nrows=1, ncols=2, figsize=(10, 5))
-        ax = axr.reshape(
-            -1,
-        )
-        plt.sca(ax[0])
-        plt.imshow(target_slice, cmap="gray")
-        plt.sca(ax[1])
-        plt.imshow(test_slice, cmap="gray")
-        plt.show()
-
-    # compute yx alignment
-    yx_offsetlist, yx_rawmeanoffset = compute_slice_alignment(
-        target_slice, test_slice, ploton=False, verbose=verbose
-    )
-
-    # specify that there is 0 offset in Z currently because that has not been identified.
-    meanoff_yx = list([0]) + list(yx_rawmeanoffset)
-    if verbose:
-        print("meanoff_in", meanoff_yx)
-
-    # now align the images based on the offset determined above
-    in_for_match = [
-        target_img_8.copy(),
-        test_img_8.copy(),
-    ]  # list of images to be matched
-    offset_list1 = [list(np.round(meanoff_yx))]  # list of offsets
-
-    # now match the images and return the matches
-    match_listxy = return_aligned_img_list_new(
-        in_for_match, offset_list1, verbose=verbose
-    )
-
-    # returned matches
-    target_img_matched_8 = match_listxy[0]
-    test_img_matched_8 = match_listxy[1]
-
-    # compute xz projections to be used for z alignment
-    target_slice = target_img_8.copy().max(axis=1)
-    test_slice = test_img_8.copy().max(axis=1)
-
-    # get y-max projections (zx images) for z alignment
-    zx_offsetlist, zx_rawmeanoffset = compute_slice_alignment(
-        target_slice,
-        test_slice,
-        ax1factor=1.5,
-        ax2factor=3,
-        nxsteps=5,
-        nysteps=5,
-        ploton=False,
-        verbose=False,
-    )
-
-    meanoffset = [zx_rawmeanoffset[0]] + list(yx_rawmeanoffset)
-
-    in_for_match_zx = [target_img_8.copy(), test_img_8.copy()]
-    offset_list_zx = [list(np.round(meanoffset))]
-
-    if verbose:
-        print("meanoffset", meanoffset)
+parser = argparse.ArgumentParser()
+
+parser.add_argument(
+    "--dfall_csv_dir",
+    required=True,
+    type=str,
+    help="csv file",
+)
+
+parser.add_argument(
+    "--barcode", required=True, type=str, help="specify barcode to analyze"
+)
+
+parser.add_argument(
+    "--output_path",
+    type=str,
+    required=True,
+    help="output dir of all processing steps. This specifies where to find the yml_configs too",
+)
+
+parser.add_argument(
```

Comment:
same comment as other file. This should accept a 1. a single position, 2 a list of positions or 3. no argument (in which case all positions will be processed).

Reply:
see above